File: RootServer/network/handler/runlayerhandler.js
const netmailbaler = require('../netmailbaler');
class RunLayerHandler {
    constructor() {
        console.log("Init RunLayerHandler");
    }
    handle(obj) {
        let name = obj.name;
        if (name == "link") {
            return this.link(obj.idA, obj.idB, obj.imuphyportA, obj.imuphyportB);
        } else if (name == "linkbroadcastresp") {
            return this.linkbroadcastresp(obj.code);
        } else if (name == "unlink") {
            return this.unlink(obj.idA, obj.idB, obj.imuphyportA, obj.imuphyportB);
        } else if (name == "unlinkbroadcastresp") {
            return this.unlinkbroadcastresp(obj.code);
        } else {
            return false;
        }
    }
    link(idA, idB, imuphyportA, imuphyportB) {
        console.log("link " + imuphyportA);
        netmailbaler.send(netmailbaler.linkbroadcast(netmailbaler.netCodeRecorder.REPEATLANDING));
        netmailbaler.send(netmailbaler.unlinkbroadcast(netmailbaler.netCodeRecorder.SUCCESS));
        return false;
    }
    linkbroadcastresp(code) {
        console.log("linkbroadcastresp " + code)
        return false;
    }
    unlink(idA, idB, imuphyportA, imuphyportB) {
        console.log("unlink " + imuphyportA)
        return false;
    }
    unlinkbroadcastresp(code) {
        console.log("unlinkbroadcastresp " + code)
        return false;
    }
}
module.exports = new RunLayerHandler();

File: ApplicationClient/WebSimulation/Assets/Resources/Scripts/network/Handler/RunLayerHandler.cs
using UnityEngine;
public class RunLayerHanlder
{
    private static RunLayerHanlder instance;
    public static RunLayerHanlder GetInstance()
    {
        if (instance == null)
        {
            instance = new RunLayerHanlder();
        }
        return instance;
    }
    private RunLayerHanlder() { }
    public bool Handle(JSONObject jb)
    {
        string name = jb["name"].str;
        if(name == "linkbroadcast")
        {
            return LinkBroadcast(jb["code"].n);
        }
        else if(name == "unlinkbroadcast")
        {
            return UnLinkBroadcast(jb["code"].n);
        }
        else
        {
            return false;
        }
    }

    private bool LinkBroadcast(float code)
    {
        Debug.Log("LinkBroadcast" + code);
        return false;
    }

    private bool UnLinkBroadcast(float code)
    {
        Debug.Log("UnLinkBroadcast" + code);
        return false;
    }
}
